File: ThreadLab4Java/ThreadLab4/src/Table.java
import java.util.concurrent.Semaphore;

public class Table {
    private final Semaphore[] forks = new Semaphore[5];

    public Table() {
        for (int i = 0; i < forks.length; i++) {
            forks[i] = new Semaphore(1);
        }
    }

    public void getFork(int id) {
        try {
            forks[id].acquire();
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
        }
    }

    public void putFork(int id) {
        forks[id].release();
    }
}

File: ThreadLab4C#/ThreadLab4/ThreadLab4/Table.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ThreadLab4
{
    public class Table
    {
        private readonly Semaphore[] _forks = new Semaphore[5];

        public Table()
        {
            for (int i = 0; i < _forks.Length; i++)
            {
                _forks[i] = new Semaphore(1, 1);
            }
        }

        public void GetFork(int id)
        {
            _forks[id].WaitOne();

        }

        public void PutFork(int id)
        {
            _forks[id].Release();
        }

    }
}
